### PR TITLE
Fred forum misc type

### PR DESCRIPTION
**Pull request by fredpaist**
issue #1525

### CODE REVIEW

File: routes/web.php [hunk 1]
```diff
@@ -115,6 +115,24 @@
...
+
+Route::post('forum/store/vaba-teema', 'V2MiscController@store')
+    ->name('forum.store.misc')
+    ->middleware('role:regular');
+
+Route::get('forum/vaba-teema/{content}/edit', 'V2MiscController@edit')
+    ->name('forum.edit.misc')
+    ->middleware('role:regular');
+
+Route::post('forum/vaba-teema/{content}/update', 'V2MiscController@store')
```
Comment (kristjanjansen) [inline routes/web.php]: V2MiscController@update

File: app/Http/Controllers/V2MiscController.php [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+            ->with('footer', region('Footer'))
+
+            ->render();
+    }
+
+    public function store(Request $request, Content $content = null)
+    {
+        $this->validate($request, $this->rules);
+
+        return $content->id ? $this->update($request, $content) : $this->newContent($request, $request->type);
```
Comment (kristjanjansen) [inline app/Http/Controllers/V2MiscController.php]: Lets just have storing stuff here, do not try store and update in same go. copy the ```newContent``` stuff here

File: app/Http/Controllers/V2MiscController.php [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    }
+
+    public function newContent($request, $type)
```
Comment (kristjanjansen) [inline app/Http/Controllers/V2MiscController.php]: move this to store

File: app/Http/Controllers/V2MiscController.php [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+        ]);
+
+        return redirect()
+            ->route(''.$type.'.index')
+            ->with('info', trans('content.store.status.'.config("content_$type.store.status", 1).'.info', [
+                'title' => $content->title,
+            ]));
+    }
+
+    public function edit(Content $content)
```
Comment (kristjanjansen) [inline app/Http/Controllers/V2MiscController.php]: What is passed here? id or are you using https://laravel.com/docs/5.4/routing#route-model-binding ? We do not (yet) use it elsewere. I'd use passing id's like we always have done https://github.com/tripikad/trip2/blob/master/app/Http/Controllers/V2BlogController.php#L193 etc and create a new PR in the future to use route model binding everywhere

File: routes/web.php [hunk 5]
```diff
@@ -115,6 +115,24 @@
...
+
+Route::get('forum/create/vaba-teema', 'V2MiscController@create')
+    ->name('forum.create.misc')
+    ->middleware('role:regular');
+
+Route::post('forum/store/vaba-teema', 'V2MiscController@store')
+    ->name('forum.store.misc')
+    ->middleware('role:regular');
+
+Route::get('forum/vaba-teema/{content}/edit', 'V2MiscController@edit')
```
Comment (kristjanjansen) [inline routes/web.php]: Lets use id for now. See V2MiscController@edit comment why

File: config/content_misc.php [hunk 6]
```diff
@@ -0,0 +1,56 @@
+<?php
```
Comment (kristjanjansen) [inline config/content_misc.php]: Do we really need this stuff?

File: resources/lang/et/content.php [hunk 7]
```diff
@@ -274,6 +274,10 @@
 
     'buysell.no.results' => 'Hetkel pole ühtegi postitust rubriigis "Ost-müük".',
 
+    //Misc
+
+    'misc.index.title' => 'Vaba teema',
+    'misc.create.title' => 'Lisa uus teema',
```
Comment (kristjanjansen) [inline resources/lang/et/content.php]: Lisa vaba teema postitus

File: config/content_misc.php (hunk 6 above)
Comment (fredpaist) [inline config/content_misc.php]: Yes, could not create comment without it right now, gives an error "Division by zero".

**Comment by kristjanjansen** [comment]
- misc/show: right url to create new post
- convert: keep other topics, remove destination
- add topic forms to create / edit

**Comment by kristjanjansen** [comment]
Notes:

misc index notes

title = Vaba teema (tripikad raagivad asemel)
body = postitused ja arutelud muudel teemadel


misc create/edit notes

title = misiganessealpraeguon
body = Siia postita mõtteid muudel teemadel

**Pull request merged**